components/board/genericlinux/board.go

```diff
@@ -276,23 +276,29 @@ func (b *Board) reconfigureAnalogs(ctx context.Context, newConf *LinuxBoardConfi
 		}
 
 		stillExists[c.Name] = struct{}{}
-		if curr, ok := b.analogs[c.Name]; ok {
+		if curr, ok := b.analogReaders[c.Name]; ok {
 			if curr.chipSelect != c.ChipSelect {
 				ar := &board.MCP3008AnalogReader{channel, bus, c.ChipSelect}
-				curr.reset(ctx, curr.chipSelect, board.SmoothAnalogReader(ar, c, b.logger))
+				curr.reset(ctx, curr.chipSelect,
+					board.SmoothAnalogReader(ar, board.AnalogConfig{
+						AverageOverMillis: c.AverageOverMillis, SamplesPerSecond: c.SamplesPerSecond,
+					}, b.logger))
 			}
 			continue
 		}
 		ar := &board.MCP3008AnalogReader{channel, bus, c.ChipSelect}
-		b.analogs[c.Name] = newWrappedAnalog(ctx, c.ChipSelect, board.SmoothAnalogReader(ar, c, b.logger))
+		b.analogReaders[c.Name] = newWrappedAnalogReader(ctx, c.ChipSelect,
+			board.SmoothAnalogReader(ar, board.AnalogConfig{
```

Comment:
hmm, I wonder how expandable the analog smoother config parameters are to new board models.  They may also be tied to the MCP3008 reader. Is the numato using them in any way?

Reply:
The numato does use the smoother. the analog smoother only uses the parameters `SamplesPerSecond` and `AverageOverMillis` so I was planning on including those parameters in the new common board analog config, and then can just convert to the common config to call the smoother in generic linux.